### PR TITLE
feat(service-principal): Create terragrunt to deploy service princpal

### PR DESCRIPTION
**Pull request by leohauschild**

### CODE REVIEW

File: infrastructure/azure/eastus/service-principal/terragrunt.hcl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+locals {
+  az-eastus-constants = yamldecode(file("../az-eastus-constants.yaml"))
+  global-constants = yamldecode(file("../../../global-constants.yaml"))
+}
+
+inputs = {
+  app_name = "conquerproject-gitops"
+  
+  subjects = [
+    "repo:conquer-project/conquer_plataform:ref:refs/heads/main",
```
Comment (benhur-araujo) [inline infrastructure/azure/eastus/service-principal/terragrunt.hcl:10]:
```suggestion
    "repo:conquerproject/conquer-plataform:ref:refs/heads/main",
```

File: infrastructure/azure/eastus/service-principal/terragrunt.hcl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "repo:conquer-project/conquer_plataform:pull_request",
```
Comment (benhur-araujo) [inline infrastructure/azure/eastus/service-principal/terragrunt.hcl:11]:
```suggestion
    "repo:conquerproject/conquer-plataform:pull_request", 
```

File: infrastructure/azure/eastus/service-principal/terragrunt.hcl [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  permissions = {
+     "/subscriptions/907cde31-ecc8-45c6-adb3-3c86455c885b" = [
+      "Contributor",
+     ],
+
+  }
+}
+
+terraform {
+  source = "/home/lhausch/my-repos/conquerproject/tf-modules/service-principal"
```
Comment (benhur-araujo) [inline infrastructure/azure/eastus/service-principal/terragrunt.hcl:23]: point to tf-modules repository

File: infrastructure/azure/eastus/service-principal/terragrunt.hcl [hunk 4]
```diff
@@ -0,0 +1,28 @@
+locals {
+  az-eastus-constants = yamldecode(file("../az-eastus-constants.yaml"))
+  global-constants = yamldecode(file("../../../global-constants.yaml"))
+}
+
+inputs = {
+  app_name = "conquerproject-gitops"
+
```
Comment (benhur-araujo) [inline infrastructure/azure/eastus/service-principal/terragrunt.hcl:8]:
```suggestion
owners = local.global-constants.maintainers  
```

File: infrastructure/azure/eastus/service-principal/terragrunt.hcl [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  global-constants = yamldecode(file("../../../global-constants.yaml"))
+}
+
+inputs = {
+  app_name = "conquerproject-gitops"
+ 
+  owners = local.global-constants.maintainers
+
+  subjects = [
+    "conquerproject-conquer-plataform-main", #<name>-org-repo-main
```
Comment (benhur-araujo) [inline infrastructure/azure/eastus/service-principal/terragrunt.hcl:12]: this format is weird, it will work?

Comment (leohauschild) [reply infrastructure/azure/eastus/service-principal/terragrunt.hcl]: it worked for my test, now I fixed it and I think everything is working well

**Review by benhur-araujo — APPROVED** [review]

**Pull request merged**